src/main/scala/is/hail/io/LoadMatrix.scala

```diff
@@ -306,8 +330,23 @@ object LoadMatrix {
         }
       }
 
+    val rowKeys: RDD[RegionValue] = rdd.mapPartitions { it =>
```

Comment:
You only need to pull out the row keys if it is faster than just pulling out the keys from the original RDD.  If you pass fastKeys = None, OrderedRVD will do this.